src/main/java/com/ss/lms/controller/BorrowerController.java

```diff
@@ -1,187 +1,116 @@
 import com.ss.lms.entity.Book;
 import com.ss.lms.entity.Borrower;
 import com.ss.lms.entity.Branch;
 import com.ss.lms.entity.Copies;
 import com.ss.lms.entity.Loans;
 import com.ss.lms.service.BorrowerService;
-import com.ss.lms.service.ConnectionUtil;
-
-
 
 @RestController
+@RequestMapping("borrower/")
 public class BorrowerController {
-	
+
 	@Autowired
 	BorrowerService borrowerService;
-	
-	@Autowired
-	Branch branch;
-	
-	
 
-	@RequestMapping(path = "borrower/{cardNo}/library/{branchId}/books/{bookId}:checkout", method = RequestMethod.POST, produces = {
-					MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
+
+	@PostMapping(path = "{cardNo}/branches/{branchId}/books/{bookId}:checkout", produces = {
+			MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
 	public ResponseEntity<Loans> checkOutBook(@PathVariable("cardNo") Integer cardNo,
-            @PathVariable("branchId") Integer branchId,
-            @PathVariable("bookId") Integer bookId) throws SQLException {
-		
-		Borrower borrower =null;
-		Book book = null;
-		Branch branch =null;
-		
-	
-		try {
-			
-			try {
-			
-			borrower = borrowerService.readByCardNoEssentialData (cardNo);
-			book = borrowerService.readBooksById(bookId);
-			branch = borrowerService.readBranchsById(branchId);
-			
-			Copies copy = borrowerService.readCopyByBranchIDBookID(book, branch);
-			Integer noOfCopies = copy.getNoOfCopies();
-			
-			if (noOfCopies == 0) {
-				return new ResponseEntity<Loans>(HttpStatus.NOT_FOUND);
-			}
-			}
-			catch (Exception e) {
-				return new ResponseEntity<Loans>(HttpStatus.NOT_FOUND);
-			}
-			
-			
-		}
-		catch (Exception e) {
-			return new ResponseEntity<Loans>(HttpStatus.INTERNAL_SERVER_ERROR);
+			@PathVariable("branchId") Integer branchId, @PathVariable("bookId") Integer bookId) {
+
+		Optional<Borrower> borrower = borrowerService.readBorrowerByCardNo(cardNo);
+		Optional<Book> book = borrowerService.readBooksById(bookId);
+		Optional<Branch> branch = borrowerService.readBranchsById(branchId);
+
+		if (!(borrower.isPresent() && book.isPresent() && branch.isPresent())) {
+			return new ResponseEntity<Loans>(HttpStatus.NOT_FOUND);
 		}
 
-		if (!(borrower == null || book == null || branch == null)) {
-			Loans l = null;
-			try {
-				l = borrowerService.checkOutBook(branch,book,borrower);
-				return new ResponseEntity<Loans>(l, HttpStatus.ACCEPTED);
-			} catch (Exception e) {
-				return new ResponseEntity<Loans>(l, HttpStatus.INTERNAL_SERVER_ERROR);
+		Optional<Copies> copy = borrowerService.readCopyByBranchIDBookID(bookId, branchId);
+
+		if (copy.isPresent()) {
+			Integer noOfCopies = copy.get().getNoOfCopies() != null ? copy.get().getNoOfCopies().intValue() : null;
+
+			if (noOfCopies <= 0) {
+				return new ResponseEntity<Loans>(HttpStatus.NOT_FOUND);
 			}
 
+			Loans l = borrowerService.checkOutBook(branchId, bookId, cardNo);
+			return new ResponseEntity<Loans>(l, HttpStatus.ACCEPTED);
```

Comment:
same comment as above with this. I think this should be happening on the service level.